src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private void linkLast(T value) {
+        final Node<T> last = tail;
+        final Node<T> newNode = new Node<>(last, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        checkPositionIndex(size);
+
+        Object[] obj = list.toArray();
+        int length = obj.length;
+
+        if (length == 0) {
+            return;
+        }
+
+        Node<T> prev;
+        prev = tail;
+
+        for (Object object : obj) {
+            T value = (T) object;
+            Node<T> newNode = new Node<>(prev, value, null);
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            prev = newNode;
+        }
+
+        tail = prev;
+        size += length;
```

Comment:
you can do your code better if you use method add(T value) for each element in your Object[] obj = new list.toArray()

Reply:
I am not sure that I know how to do that

Reply:
you can do that, just try, in other case you can return to this version of your code